core/anvil/src/region/mod.rs

```diff
@@ -31,62 +31,232 @@ const DATA_VERSION: i32 = 1631;
 /// Length, in bytes, of a sector.
 const SECTOR_BYTES: usize = 4096;
 
-/// The offset for each heightmap value
-const HEIGHTMAP_OFFSET: i64 = 9;
-
 /// Represents the data for a chunk after the "Chunk [x, y]" tag.
 #[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "PascalCase")]
 pub struct ChunkRoot {
-    #[serde(rename = "Level")]
     level: ChunkLevel,
-    #[serde(rename = "DataVersion")]
     data_version: i32,
 }
 
 /// Represents the level data for a chunk.
 #[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "PascalCase")]
 pub struct ChunkLevel {
     // TODO heightmaps, etc.
     #[serde(rename = "xPos")]
     x_pos: i32,
     #[serde(rename = "zPos")]
     z_pos: i32,
-    #[serde(rename = "Sections")]
     sections: Vec<LevelSection>,
-    #[serde(rename = "Biomes")]
     biomes: Vec<i32>,
-    #[serde(rename = "Entities")]
     entities: Vec<EntityData>,
     #[serde(rename = "TileEntities")]
     block_entities: Vec<BlockEntityData>,
-    #[serde(rename = "Heightmaps")]
-    heightmaps: Vec<i64>,
+    heightmaps: Heightmaps,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub struct Heightmaps {
+    // sometimes a few of those are missing, but we regenerate them anyways
+    #[serde(with = "packed_u9", default)]
+    light_blocking: Vec<u16>,
+    #[serde(with = "packed_u9", default)]
+    motion_blocking: Vec<u16>,
+    #[serde(with = "packed_u9", default)]
+    motion_blocking_no_leaves: Vec<u16>,
+    #[serde(with = "packed_u9", default)]
+    ocean_floor: Vec<u16>,
+    #[serde(with = "packed_u9", default)]
+    world_surface: Vec<u16>,
+}
+
+impl Heightmaps {
+    // TODO remove this and use hematite-nbt serde for serialization
+    pub(crate) fn pack_u9(u9_array: Vec<u16>) -> Vec<i64> {
+        let mut packed = Vec::with_capacity(36);
```

Comment:
Could the initial capacity be derived from the length of the input, instead of using this magic number?  (Or at least extract it into a constant, since it's used a few more times in this file.)